dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/utils/ProxySupportUtil.java

```diff
@@ -33,13 +36,45 @@ public class ProxySupportUtil {
     private static URL extractURL(HttpServletRequest request) {
         String urlString;
         String refererHeader = request.getHeader(HTTP_HEADER_REFERER);
+
+        String docPath = "/openapi";
+        String uiPath = docPath+"/ui";
+
+        // retrieve the context from the request so that we can retrieve the values of the OpenAPI endpoints
+        BundleContext bundleContext = (BundleContext) request.getServletContext().getAttribute("osgi-bundlecontext");
+        //ensure we have a context in case for some reason this is activated with the bundle having stopped
+        if(bundleContext != null) {
+            ServiceTracker<OpenAPIEndpointProvider,OpenAPIEndpointProvider> openAPIEndpointTracker = new ServiceTracker<>(bundleContext, OpenAPIEndpointProvider.class, null);
+            openAPIEndpointTracker.open();
+            docPath = openAPIEndpointTracker.getService().getOpenAPIDocUrl();
+            uiPath = openAPIEndpointTracker.getService().getOpenAPIUIUrl();
+            openAPIEndpointTracker.close();
+        }
```

Comment:
There's no point having a `ServiceTracker` if you're going to open and immediately close it. The idea of the `ServiceTracker` is that it's long-lived and watches for changes so that we avoid having to look up the service every time.

If you don't have a long-lived object to attach the `ServiceTracker` to, just look up the service directly with [BundleContext.getService](https://docs.osgi.org/javadoc/osgi.core/7.0.0/org/osgi/framework/BundleContext.html#getService-org.osgi.framework.ServiceReference-).

Whether you use a tracker or `BundleContext.getService`, you still need to check whether the returned service is `null`.